apps/api/v2/src/modules/workflows/inputs/create-workflow.input.ts

```diff
@@ -0,0 +1,229 @@
+import { ApiProperty, ApiPropertyOptional } from "@nestjs/swagger";
+import { Type } from "class-transformer";
+import {
+  IsBoolean,
+  IsEmail,
+  IsEnum,
+  ArrayMinSize,
+  IsNumber,
+  IsOptional,
+  IsString,
+  ValidateNested,
+  ValidateIf,
+} from "class-validator";
+
+export enum WorkflowTriggerType {
+  BEFORE_EVENT = "BEFORE_EVENT",
+  EVENT_CANCELLED = "EVENT_CANCELLED",
+  NEW_EVENT = "NEW_EVENT",
+  AFTER_EVENT = "AFTER_EVENT",
+  RESCHEDULE_EVENT = "RESCHEDULE_EVENT",
+}
+
+export enum WorkflowTimeUnit {
+  HOUR = "HOUR",
+  MINUTE = "MINUTE",
+  DAY = "DAY",
+}
+
+export enum RecipientType {
+  HOST = "HOST",
+  ATTENDEE = "ATTENDEE",
+  EMAIL = "EMAIL",
+  PHONE_NUMBER = "PHONE_NUMBER",
+}
+
+export enum StepAction {
+  EMAIL_HOST = "EMAIL_HOST",
+  EMAIL_ATTENDEE = "EMAIL_ATTENDEE",
+  EMAIL_ADDRESS = "EMAIL_ADDRESS",
+  SMS_ATTENDEE = "SMS_ATTENDEE",
+  SMS_NUMBER = "SMS_NUMBER",
+  WHATSAPP_ATTENDEE = "WHATSAPP_ATTENDEE",
+  WHATSAPP_NUMBER = "WHATSAPP_NUMBER",
+}
+
+export enum TemplateType {
```

Comment:
First, can we switch from enum to type so that we don't have to import enum when writing test? We should use types instead of enums whenever possible.

```
export const TemplateTypes = [
  "reminder",
  "custom"
] as const;

export type TemplateType = (typeof TemplateTypes)[number]
```

Second, can we have lowercase format and also for `StepAction` have `smsAttendee` instead of "SMS_ATTENDEE" to follow other endpoint standards.